Code/Framework/AzToolsFramework/AzToolsFramework/Asset/AssetUtils.cpp

```diff
@@ -241,8 +241,24 @@ namespace AzToolsFramework::AssetUtils
         return configFiles;
     }
 
-    bool UpdateFilePathToCorrectCase(AZStd::string_view rootPath, AZStd::string& relPathFromRoot)
+    bool UpdateFilePathToCorrectCase(AZStd::string_view rootPath, AZStd::string& relPathFromRoot, bool checkEntirePath /* = true */)
     {
+        // The reason this is so expensive is that it could also be the case that the DIRECTORY path is the wrong case.
+        // For example, the actual path might be /SomeFolder/textures/whatever/texture.dat
+        // but the real file on disk is actually /SomeFolder/Textures/Whatever/texture.dat 
+        // Note the case difference of the directories.  If you were to ask the operating system just to list all of
+        // the files in the input folder (/SomeFolder/textures/whatever/*) it would not find any since the directory 
+        // does not itself exist.  Not only that, but many operating system calls on case-insensitive file systems
+        // actually use the input given in their return - that is, if you ask a WINAPI call to construct an absolute path
+        // to a file and give it the wrong case as input, the output absolute path will also be the wrong case.
+        // Thus, to actually correct a path it has to start at the bottom and whenever it encounters
+        // a path segment, it has to do a read of the actual directory information to see which files exist in that
+        // directory, and whether a file or directory exists with the correct name but with different case.
+
+        // if checkEntirePath is false, we only check the file name, and nothing else.  This is for the case where
+        // the path is known to be good except for the file name, such as when the relPathFromRoot comes from just
+        // taking an existing, known-good file and replacing its extension only.
```

Comment:
Interesting - in this case, I'm not sure its quite necessary until we rearchitect the thing, since the system calling _into_ this function does have state, and does have its own case insensitive hash - its only calling this function when it encounters a case insensitive hash miss to make sure that what it then adds into the cache is authoritatively the correct case.

Reply:
I assume this is not something we need to fix for point release before this goes in?

Reply:
Well there is one low-hanging fruit, which we could tackle if we need to improve the performance in the future even more.
That is the "potential" repeated calls to `FileIOBase::FindFiles`. Since that could be called multiple times for the same directory, that directory information with the correct case and case-insensitive directory hash could be a quick win without storing too much data in-memory.

However we can wait on that.